### DISCUSSION TITLE
Swapped Google for Mapbox, unable to publish meetings

### DISCUSSION THREAD

**Discussion by jessfrick** [code4recovery/12-step-meeting-list#1362] [General]
Greetings from Central Florida! We operate our website at https://cflintergroup.org

WordPress Version: 6.4.3
PHP Version: 8.1
12 Step Meeting List: Version 3.14.32

We recently updated our site and wanted to employ the lovely TSML UI, which meant we needed to switch our keys from Google to Mapbox. I presume it's expected behavior that entering one key/token removes the other, and that's what's happened here. 

Since making this change, I'm now unable to publish new meetings. 

I checked my console and saw the following: 

![Edit Meeting “APOPKA BIG BOOK” ‹ Central Florida Intergroup — WordPress 2024-02-18 at 11 12 48 PM](https://github.com/code4recovery/12-step-meeting-list/assets/6118303/0db1d9b9-d050-42d4-a57e-ec342d74d6ff)

I'd already cleared all caches and wasn't sure what was still calling the Google Maps API, but I tried doing it again, re-adding my Mapbox token and cleared the console errors. The second time, I got this: 

![Edit Meeting “APOPKA BIG BOOK” ‹ Central Florida Intergroup — WordPress 2024-02-18 at 11 32 27 PM](https://github.com/code4recovery/12-step-meeting-list/assets/6118303/6edeef69-085c-4f41-85d8-de4482f1463d)

I tried reverting to the Legacy UI with Google Maps, but still am unable to save a draft or publish a meeting. The site is back to Mapbox and is set the way I'd like it to be, only with a working publish button.  ;) 

I read through a few older posts that referenced similar challenges (#683, #480), but it seems the steps used to solve those are now outdated. Of note, I should mention that the site has just been move to Pressable, which operates on WP Cloud, same as WordPress.com. I don't believe the issues that came up in #1257 are related, but just in case, I wanted to share this. 

Separate but related ask: is there a good reason to keep the feed restricted? Or does it not matter? Will it cause a blip in the feed if I change from protected to public? 

Thank you in advance for any help you can offer!

**Comment by tim-rohrer** [comment]
Hi Jess,

Thanks for the extensive research you've done through past issues. 

For the second image you have above, please try to reproduce the error and then expand the message by clicking on the arrow next to GET.

**Reply by jessfrick** [reply]
Thanks for the fast response, Tim! I was able to reproduce the error: 

![Edit Meeting “APOPKA BIG BOOK” ‹ Central Florida Intergroup — WordPress 2024-02-19 at 11 39 15 AM](https://github.com/code4recovery/12-step-meeting-list/assets/6118303/06fbffd9-b277-4a62-98f2-83e020163784)
 
I'm at a loss here. 🙃

**Comment by joshreisner** [comment]
hi @jessfrick something weird is happening when your site is trying to geocode this result. could you check your theme's `functions.php` to see if there's any code that got added, perhaps something related to geocoding?

another thing to try: set WP_DEBUG to true (temporarily) in your site's wp-config.php, and then open the link in your browser console. should start with `https://cflintergroup.org/wp-admin/admin-ajax.php?action=tsml_geocode&…` hopefully this will yield some useful info, thanks!

**Reply by jessfrick** [reply]
Hi @joshreisner ! 

My theme's `functions.php` looks totally normal: https://share.zight.com/5zuvnoAm 

I've also set WP_DEBUG to true and followed the error through the console... I'm still not seeing whatever is causing this. 

![Edit Meeting “APOPKA BIG BOOK” ‹ Central Florida Intergroup — WordPress 2024-02-19 at 5 34 28 PM](https://github.com/code4recovery/12-step-meeting-list/assets/6118303/070f00c4-6dbd-47d3-9f1d-f55963b1f8d6)

![Edit Meeting “APOPKA BIG BOOK” ‹ Central Florida Intergroup — WordPress 2024-02-19 at 5 33 21 PM](https://github.com/code4recovery/12-step-meeting-list/assets/6118303/2cf99bfb-138c-4604-be4e-a48a61c6c479)

The `load-scripts.php` link brings me here: 

![Edit Meeting “APOPKA BIG BOOK” ‹ Central Florida Intergroup — WordPress 2024-02-19 at 5 38 50 PM](https://github.com/code4recovery/12-step-meeting-list/assets/6118303/5478aee2-5637-46d4-9cf8-3b90593c4bb1)


I'm not the best dev for this sort of thing, but I'm willing to learn and eager to serve. <3

**Reply by joshreisner** [reply]
is WP_DEBUG still true? it's weird because i'm seeing a generic "there's a critical error" error when i visit this URL to get more information. https://cflintergroup.org/wp-admin/admin-ajax.php?action=tsml_info

and/or try right-clicking that link in the network tab - it should have `admin-ajax.php` in it - and paste it here? also visit the URL and see what it says - by design it is short-lived and we might not be able to see it in time

what's the address you're trying to geocode? are you able to geocode other addresses? if you were to make a new meeting for example

**Reply by jessfrick** [reply]
I left WP_DEBUG as true. No idea why it's throwing an error on that URL though -- I see it in network > preview as well. 

Here's what I see when I right click on it in the network tab: 

![Edit Meeting “APOPKA BIG BOOK” ‹ Central Florida Intergroup — WordPress 2024-02-20 at 8 22 39 AM](https://github.com/code4recovery/12-step-meeting-list/assets/6118303/63a1f35b-68ab-4e04-9082-175f51c61a97)

![Edit Meeting “APOPKA BIG BOOK” ‹ Central Florida Intergroup — WordPress 2024-02-20 at 8 22 55 AM](https://github.com/code4recovery/12-step-meeting-list/assets/6118303/87bfcbc7-a361-4717-936f-5a617a445ad3)

The address I'm trying to use is: 
![Edit Meeting “APOPKA BIG BOOK” ‹ Central Florida Intergroup — WordPress 2024-02-20 at 8 21 04 AM](https://github.com/code4recovery/12-step-meeting-list/assets/6118303/448b48b3-3063-45ca-9391-58bfb0a9c334)

When I try a different address -- one that I know to be correct and established -- I get the same result. All of this is the result of me trying to make a new meeting, but with a known and established address. No errors appear until I enter a location.

I'm genuinely befuddled. Thank you very much for helping me with this.

**Reply by jessfrick** [reply]
PS: flushed cache again. Edge cache has been disabled this whole time. Cleared all-time cookies. Still erroring. 

Sharing again with cleaner cookie records: 
![Edit Meeting “APOPKA BIG BOOK” ‹ Central Florida Intergroup — WordPress 2024-02-20 at 8 35 52 AM](https://github.com/code4recovery/12-step-meeting-list/assets/6118303/27e2b317-9c48-4c65-893f-b567c2795084)

Here's an existing meeting at that location: https://cflintergroup.org/meetings/?search=apopka+big&meeting=apopka-big-book

Interestingly, since I moved the site over and switched to TSML UI, the feedback enhancement feature also isn't showing.

**Reply by joshreisner** [reply]
ok thanks yes the address is fine ✅ 

the feedback enhancement won't show - it's not a feature that TSML UI supports currently. there is just sort of a basic email button when feedback emails are specified

the problem with editing the meetings is that the geocode is throwing an error, as is our info AJAX endpoint. i think we need to get to the bottom of that somehow, but i'm not sure what else to try other than turning WP_DEBUG on. perhaps there is more information you can find on the web to help output the PHP error? https://www.godaddy.com/help/enable-wordpress-debug-mode-to-display-errors-24013# for example

we also want you to open the link in the error console: GET https://cflintergroup.org… in a new tab so we can see the output. we can't do that from our end because the authorization "nonce" is stale, so it needs to be done in the moment. 

(i'm guessing right now it will show the generic "there's been a critical error" message, which is why we have to get that debug output working)

**Reply by jessfrick** [reply]
Hi there! Although my old site was on PHP 8.1 and seemed to work fine, I rolled back this one to PHP 7.4 and now it's working. 

I asked a colleague to take a look at this and he shared the following with me: 

> So, there's a fatal from a few hours ago but I don't see a recent one.
`PHP Fatal error: Uncaught TypeError: array_slice(): Argument #1 ($array) must be of type array, bool given in /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/ajax.php:24 Stack trace: #0 /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/ajax.php(24): array_slice(false, 0, 25) #1 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(324): tsml_ajax_info('') #2 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array) #3 /wordpress/core/6.4.3/wp-includes/plugin.php(517): WP_Hook->do_action(Array) #4 /wordpress/core/6.4.3/wp-admin/admin-ajax.php(203): do_action('wp_ajax_nopriv_...') #5 {main} thrown in /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 24`

> Which is a PHP 8+ conflict. And... 
`PHP Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, bool given in /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/functions.php:658 Stack trace: #0 /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/ajax.php(328): tsml_geocode('201 S Park Ave,...') #1 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(324): tsml_ajax_geocode('') #2 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array) #3 /wordpress/core/6.4.3/wp-includes/plugin.php(517): WP_Hook->do_action(Array) #4 /wordpress/core/6.4.3/wp-admin/admin-ajax.php(188): do_action('wp_ajax_tsml_ge...') #5 {main} thrown in /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/functions.php on line 658
`

> Both seem related to what you're encountering. 

It's interesting that rolling back to 7.4 has corrected whatever was causing the issue. Eventually we will need to upgrade to the latest supported version, but Automattic (Pressable's parent company) is providing patches for 7.4 for now, so we have time. 

With this information, do you have any suggestions on things I can do to best prepare for this?

**Reply by JoshuaGoode** [reply]
Discussed things with @jessfrick a bit and identified the following fatal errors triggering due to TypeError conflicts with PHP 8+.

The temporary workaround I recommended for now was to downgrade to PHP 7.4 which seems to have things working a bit for now.

```
PHP Fatal error: Uncaught TypeError: array_slice(): Argument #1 ($array) must be of type array, bool given in /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/ajax.php:24 Stack trace: #0 /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/ajax.php(24): array_slice(false, 0, 25) #1 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(324): tsml_ajax_info('') #2 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array) #3 /wordpress/core/6.4.3/wp-includes/plugin.php(517): WP_Hook->do_action(Array) #4 /wordpress/core/6.4.3/wp-admin/admin-ajax.php(203): do_action('wp_ajax_nopriv_...') #5 {main} thrown in /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 24
```

```
PHP Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, bool given in /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/functions.php:658 Stack trace: #0 /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/ajax.php(328): tsml_geocode('201 S Park Ave,...') #1 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(324): tsml_ajax_geocode('') #2 /wordpress/core/6.4.3/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array) #3 /wordpress/core/6.4.3/wp-includes/plugin.php(517): WP_Hook->do_action(Array) #4 /wordpress/core/6.4.3/wp-admin/admin-ajax.php(188): do_action('wp_ajax_tsml_ge...') #5 {main} thrown in /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/functions.php on line 658
```

**Reply by joshreisner** [reply]
excellent, thank you!! we'll look into this and get back to you
